virtual_rainforest/models/abiotic/energy_balance.py

```diff
@@ -0,0 +1,321 @@
+        "Vq": 1,  # Volumetric quartz content of soil
+        "Vm": 1,  # Volumetric mineral content of soil
+        "Vo": 1,  # Volumetric organic content of soil
+        "Mc": 1,  # Mass fraction of clay
+        "rho": 1,  # Soil bulk density (Mg / m^3)
+        "b": 1,  # Shape parameter for Campbell model (dimensionless, > 1)
+        "psi_e": 1,  # Matric potential (J / m^3)
+    },
+    "Clay": {
+        "Smax": 0.1,  # Volumetric water content at saturation (m^3 / m^3)
+        "Smin": 0.1,  # Residual water content (m^3 / m^3)
+        "alpha": 0.1,  # Shape parameter of the van Genuchten model (cm^-1)
+        "n": 0.1,  # Pore_size_distribution_parameter (dimensionless, > 1)
+        "Ksat": 0.1,  # Saturated hydraulic conductivity (cm / day)
+        "Vq": 0.1,  # Volumetric quartz content of soil
+        "Vm": 0.1,  # Volumetric mineral content of soil
+        "Vo": 0.1,  # Volumetric organic content of soil
+        "Mc": 0.1,  # Mass fraction of clay
+        "rho": 0.1,  # Soil bulk density (Mg / m^3)
+        "b": 0.1,  # Shape parameter for Campbell model (dimensionless, > 1)
+        "psi_e": 0.1,  # Matric potential (J / m^3)
+    },
+}
+
+
+class EnergyBalance:
+    """EnergyBalance method.
+
+    Attributes:
+        air_temperature: NDArray[np.float32],
+        soil_temperature: NDArray[np.float32],
+        canopy_temperature: NDArray[np.float32],
+        temperature_above_canopy: NDArray[np.float32],
+        canopy_wind_speed: NDArray[np.float32],
+        canopy_node_heights: NDArray[np.float32],
+        soil_node_depths: NDArray[np.float32],
+        height_of_above_canopy: NDArray[np.float32],
+        relative_humidity_2m: NDArray[np.float32],
+        air_temperature_2m: NDArray[np.float32],
+        mean_annual_temperature: NDArray[np.float32], temperature of deepest soil layer
+        atmospheric_pressure_2m: NDArray[np.float32],
+        absorbed_radiation: : NDArray[np.float32],
+        air_conductivity: NDArray[np.float32],
+        leaf_conductivity: NDArray[np.float32],
+        air_leaf_conductivity: NDArray[np.float32],
+        sensible_heat_flux: NDArray[np.float32],
+        latent_heat_flux: NDArray[np.float32],
+        ground_heat_flux: NDArray[np.float32],
+        diabatic_correction_factor: NDArray[np.float32]
+        soil_type: NDArray[np.string_]
+        soil_depth: NDArray[np.float32]
+        soil_moisture: NDArray[np.float32]
+        soil_moisture_reference: NDArray[np.float32]
+        wind_profile: NDArray[np.float32]
+    """
+
+    def __init__(
+        self,
+        data: dict[str, float],
+        soil_type: NDArray[np.string_],
+        soil_depth: NDArray[np.float32] = np.array(2.0, dtype=float),
+        mean_annual_temperature: NDArray[np.float32] = np.array(20, dtype=float),
+        canopy_layers: NDArray[np.int32] = np.array(3, dtype=int),
+        soil_layers: NDArray[np.int32] = np.array(2, dtype=int),
```

Comment:
Are `canopy_layers` and `soil_layers` supposed to be arrays? You seem to be using them as scalars below and I don't think the code will work if actual arrays are passed in.

Reply:
Does `soil_type` need to be an array either?

Reply:
The initial idea was that canopy layers could be different across cells, but as it will be the same everywhere, yes, it should be a scalar. Soil type differs between grid cells